frontend/app/routes/_gcweb-app.personal-information.mailing-address.edit.tsx

```diff
@@ -132,67 +179,101 @@ export default function PersonalInformationMailingAddressEdit() {
-          name="city"
-          required
-          key={isCopyAddressChecked ? homeAddressInfo?.city : addressInfo?.city}
-          defaultValue={defaultValues.city}
-          errorMessage={errorMessages.city}
-        />
-        <InputField
-          id="province"
-          label={t('personal-information:mailing-address.edit.field.province')}
-          name="province"
-          key={isCopyAddressChecked ? homeAddressInfo?.province : addressInfo?.province}
-          defaultValue={defaultValues.province}
-          errorMessage={errorMessages.province}
-        />
-        <InputField
-          id="postalCode"
-          label={t('personal-information:mailing-address.edit.field.postal-code')}
-          name="postalCode"
-          key={isCopyAddressChecked ? homeAddressInfo?.postalCode : addressInfo?.postalCode}
-          defaultValue={defaultValues.postalCode}
-          errorMessage={errorMessages.postalCode}
-        />
-        <InputField
-          id="country"
-          label={t('personal-information:mailing-address.edit.field.country')}
-          name="country"
-          required
-          key={isCopyAddressChecked ? homeAddressInfo?.country : addressInfo?.country}
-          defaultValue={defaultValues.country}
-          errorMessage={errorMessages.country}
-        />
-
-        <div className="flex flex-wrap gap-3">
-          <button id="change-button" className="btn btn-primary btn-lg">
-            {t('personal-information:mailing-address.edit.button.change')}
-          </button>
-          <Link id="cancel-button" to="/personal-information" className="btn btn-default btn-lg">
-            {t('personal-information:mailing-address.edit.button.cancel')}
-          </Link>
-        </div>
-      </Form>
+      {isCopyAddressChecked && (
+        <Form className="max-w-prose" method="post">
+          <p>
+            <strong>{t('personal-information:mailing-address.edit.copy-home-address-note')}</strong>
+          </p>
+          <Address
+            address={homeAddressInfo?.address ?? ''}
+            city={homeAddressInfo?.city ?? ''}
+            provinceState={regionList.find((region) => region.code === homeAddressInfo?.province)?.nameEn}
+            postalZipCode={homeAddressInfo?.postalCode ?? ''}
+            country={countryList.find((country) => country.code === homeAddressInfo?.country)?.nameEn ?? ' '}
+          />
+          <input type="text" id="address" name="address" value={defaultValues.address} readOnly hidden />
+          <input type="text" id="city" name="city" value={defaultValues.city} readOnly hidden />
+          <input type="text" id="province" name="province" value={defaultValues.province} readOnly hidden />
+          <input type="text" id="postalCode" name="postalCode" value={defaultValues.postalCode} readOnly hidden />
+          <input type="text" id="country" name="country" value={defaultValues.country} readOnly hidden />
```

Comment:
Avoid using `hidden` and `readonly` inputs as they can be tampered with by users. Alternatively, we can see if the checkbox is checked in the `action` and if so, set the home address as the new address in the session.

Reply:
fixed